### ISSUE TITLE
Build failed

### ISSUE THREAD

**Issue by weaversa** [GaloisInc/cryptol#939]
This worked on Monday :-)

```
Failed to build cryptol-2.9.1.99.
Build log (
/home/cryptol/.cabal/logs/ghc-8.10.2/cryptol-2.9.1.99-cdcac95f606a4f5adc6b532a80d45f6603beadd0755d5065ea90548e6c31c3e5.log
):
Configuring library for cryptol-2.9.1.99..
Preprocessing library for cryptol-2.9.1.99..
Building library for cryptol-2.9.1.99..
[  1 of 103] Compiling Cryptol.AES      ( src/Cryptol/AES.hs, dist/build/Cryptol/AES.o, dist/build/Cryptol/AES.dyn_o )
[  2 of 103] Compiling Cryptol.Backend.Arch ( src/Cryptol/Backend/Arch.hs, dist/build/Cryptol/Backend/Arch.o, dist/build/Cryptol/Backend/Arch.dyn_o )
[  3 of 103] Compiling Cryptol.Backend.What4.SFloat ( src/Cryptol/Backend/What4/SFloat.hs, dist/build/Cryptol/Backend/What4/SFloat.o, dist/build/Cryptol/Backend/What4/SFloat.dyn_o )
[  4 of 103] Compiling Cryptol.ModuleSystem.Fingerprint ( src/Cryptol/ModuleSystem/Fingerprint.hs, dist/build/Cryptol/ModuleSystem/Fingerprint.o, dist/build/Cryptol/ModuleSystem/Fingerprint.dyn_o )
[  5 of 103] Compiling Cryptol.Prelude  ( src/Cryptol/Prelude.hs, dist/build/Cryptol/Prelude.o, dist/build/Cryptol/Prelude.dyn_o )

src/Cryptol/Prelude.hs:35:35: error:
    * Exception when trying to run compile-time code:
        lib/Cryptol/Reference.cry: openFile: does not exist (No such file or directory)
      Code: template-haskell-2.16.0.0:Language.Haskell.TH.Quote.quoteExp
              there "lib/Cryptol/Reference.cry"
    * In the quasi-quotation: [there|lib/Cryptol/Reference.cry|]
   |
35 | preludeReferenceContents = B.pack [there|lib/Cryptol/Reference.cry|]
   |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cabal: Failed to build cryptol-2.9.1.99 (which is required by exe:cryptol from
cryptol-2.9.1.99 and exe:cryptol-html from cryptol-2.9.1.99). See the build
log above for details.
```

**Comment by weaversa** [comment]
From master, just a few minutes ago.

**Comment by robdockins** [comment]
That's odd... the CI builds and local builds here both seem to be working.  Makes me wonder if there are some odd caching effects going on with cabal.  Maybe try to delete your `dist-newstyle` directory and rebuild?

**Comment by weaversa** [comment]
I started the build fresh in a new docker container. I’ll send you the build script in a minute

**Comment by robdockins** [comment]
I believe this is fixed by #940.  The issue here was that `cabal install` builds were failing because they build from source tarballs generated by `cabal sdist`, which in turn was not picking up the new `Cryptol::Reference` module because it was placed in a subdirectory of `lib`.  This was masked by the fact that local builds and CI generally build in place instead of from the generated source tarball.  Maybe this warrants some change in our CI procedures? I'm not sure.

At any rate, the fix was to update the cabal file to include data files even in subdirectories of `lib`.
